nuclio/config.py

```diff
@@ -74,8 +74,26 @@ def get_in(obj, keys):
     return obj
 
 
+def split_by_dots_with_escaping(key: str):
+    """
+    splits the key by dots, taking escaping into account so that the key can contain dots
+    """
+    parts = []
+    current_key, escape = "", False
+    for char in key:
+        if char == "." and not escape:
+            parts.append(current_key)
+            current_key = ""
+        elif char == "\\":
```

Comment:
what if char is `\"` ? will it catch it? and if so will it filter the `"`?
what is the expected use case here? is the escaping done by the user?

Reply:
It will take it as a regular char in the key, I added a test with a case containing `\"`, the result is:
![image](https://user-images.githubusercontent.com/62285310/216285598-51904c28-0b51-491f-a4dd-eefbcdfb4161.png)